Python/compile.c

```diff
@@ -3159,15 +3203,20 @@ compiler_try_except(struct compiler *c, stmt_ty s)
     ADDOP_JUMP_NOLINE(c, JUMP_FORWARD, orelse);
     n = asdl_seq_LEN(s->v.Try.handlers);
     compiler_use_next_block(c, except);
+
+    c->u->u_lineno = -1;
+    ADDOP_JUMP(c, SETUP_CLEANUP, cleanup);
+    ADDOP(c, PUSH_EXC_INFO);
     /* Runtime will push a block here, so we need to account for that */
     if (!compiler_push_fblock(c, EXCEPTION_HANDLER, NULL, NULL, NULL))
         return 0;
     for (i = 0; i < n; i++) {
         excepthandler_ty handler = (excepthandler_ty)asdl_seq_GET(
             s->v.Try.handlers, i);
-        if (!handler->v.ExceptHandler.type && i < n-1)
-            return compiler_error(c, "default 'except:' must be last");
         SET_LOC(c, handler);
+        if (!handler->v.ExceptHandler.type && i < n-1) {
+            return compiler_error(c, "default 'except:' must be last");
+        }
```

Comment:
Is this another case where you happened to notice it was incorrect, or is this change due to this PR?

Reply:
I think this was incorrect before. I'll see if it makes sense to apply it to 3.10b.